src/integer_mod_q/mat_zq/from.rs

```diff
@@ -125,6 +125,44 @@ impl MatZq {
         }
         out
     }
+
+    /// Generate a `num_rows` times `num_columns` matrix with `1` on the
+    /// diagonal and `0` anywhere else with a given modulus (if the modulus is `1`
+    /// every entry is `0`).
+    ///
+    /// Parameters:
+    /// - `rum_rows`: the number of rows of the identity matrix
+    /// - `num_columns`: the number of columns of the identity matrix
+    /// - `modulus`: the modulus of the matrix
+    ///
+    /// Returns a matrix with `1` across the diagonal and `0` anywhere else.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    ///
+    /// let matrix = MatZq::identity(2, 3, 3).unwrap();
+    ///
+    /// let identity = MatZq::identity(10, 10, 3).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`InvalidMatrix`](MathError::InvalidMatrix) or
+    /// [`OutOfBounds`](MathError::OutOfBounds) if the provided number of rows and columns
+    /// are not suited to create a matrix. For further information see [`MatZq::new`].
+    /// - Returns a [`MathError`] of type [`InvalidIntToModulus`](MathError::InvalidIntToModulus)
+    /// if the modulus is not greater than `0`. For further information see [`MatZq::new`].
+    pub fn identity(
+        num_rows: impl TryInto<i64> + Display + Copy,
+        num_cols: impl TryInto<i64> + Display + Copy,
+        modulus: impl Into<Z>,
+    ) -> Result<Self, MathError> {
+        let mut out = MatZq::new(num_rows, num_cols, modulus)?;
+        if out.get_mod() != Modulus::try_from_z(&Z::ONE).unwrap() {
```

Comment:
This should make it more efficient
```suggestion
        if Z::ONE != Z::from(&out.modulus) {
```